src/c++/perf_analyzer/data_loader.cc

```diff
@@ -38,6 +38,34 @@ DataLoader::DataLoader(const size_t batch_size)
 {
 }
 
+cb::Error
+DataLoader::ValidateIOExistsInModel(
+    const std::shared_ptr<ModelTensorMap>& inputs,
+    const std::shared_ptr<ModelTensorMap>& outputs,
+    const std::string& data_directory)
+{
+  if (!std::filesystem::exists(data_directory) ||
+      !std::filesystem::is_directory(data_directory)) {
+    return cb::Error(
+        "Error: Directory does not exist or is not a directory: " +
+            std::string(data_directory),
+        pa::GENERIC_ERROR);
+  }
+
+  for (const auto& file : std::filesystem::directory_iterator(data_directory)) {
+    std::string io_name = file.path().filename().string();
+    if (inputs->find(io_name) == inputs->end() &&
+        outputs->find(io_name) == outputs->end()) {
+      return cb::Error(
+          "Provided data file '" + io_name +
+              "' does not correspond to a valid model input or output.",
+          pa::GENERIC_ERROR);
```

Comment:
Typically enums are considered best practices for status codes.
Not for this PR, but can we use enum for status codes?

Reply:
@matthewkotila @debermudez @tgerdesnv can probably speak more to this. I was trying to be consistent with what PA currently does.

Reply:
Yeah, this is following pattern, but the pattern is not following best practices and would be good to prioritize a cleanup regarding this.